misc/audit.mjs

```diff
@@ -0,0 +1,297 @@
+const mapWcs = {};
+const WCS_KEYS = [ 'offerId' , 'productArrangementCode' , 'commitment' , 'term' , 'customerSegment' , 'marketSegments' , 'offerType' , 'pricePoint' ];
+const retries = new Set();
+const fetched = new Set();
+const isRelative = (url) => (url[0] == '/');
+let defaultBufferSize = BUFFER_SIZE;
+let file = '/tmp/audit.csv';
+
+const getUrlParts = ( url ) => {
+    const domain = (new RegExp(DOMAIN_REGEXP)).exec(url)[0];
+    return {
+        domain,
+        uri: url = url.substring(domain.length),
+    }
+}
+
+const fetchDocument = (url) => {
+    console.log(`fetching ${url}...`);
+    return fetch(url);
+};
+
+const getLiveUrl = (url) => url.replaceAll('.hlx.page', '.hlx.live');
+
+async function getFragmentsFromManifest(url) {
+    const fragments = [];
+    try {
+        const response = await fetch(getLiveUrl(url));
+        if(response.status == 200 && response.size > 0) {
+            const manifest = await response?.json();
+            manifest.experiences?.data?.forEach((data) => {
+                Object.values(data)
+                    .map((value) => /^https:\/\/.+/.exec(value)?.[0])
+                    .filter((value) => value != null)
+                    .map(getLiveUrl)
+                    .forEach((url) => fragments.push(url));
+            });
+        }
+    } catch (error) {
+        console.log(`Error while fetching manifest: ${error}`)
+    }
+    return fragments;
+}
+
+async function getPersonnalizationFragments(pageContent) {
+    const persoRegexp = (new RegExp(PERSO_REGEXP, 'g')).exec(pageContent);
+    if (!persoRegexp) {
+        return null;
+    }
+    const fragments = new Set();
+    const { urls } = persoRegexp.groups;
+    const urlArr = urls.replace('\s', '').split(',');
+    urlArr.forEach(async (url) => {
+        const manifestFragments = await getFragmentsFromManifest(url);
+        manifestFragments.forEach((fragment) => fragments.add(fragment));
+    });
+    return Array.from(fragments);
+}
+
+const extractLinks = (pageContent) => {
+    const result = {};
+    const linkRegexp = new RegExp(LINK_REGEXP, 'g');
+    let match;
+    while ((match = linkRegexp.exec(pageContent)) != null) {
+        const { url } = match.groups;
+        const indexEnd = match.index + match[0].length;
+        const postExcerpt = pageContent.substring(indexEnd, indexEnd + EXCERPT_SIZE).replaceAll(/[,\n\s]+/g,'');
+        Object.entries(HREF_REGEXPS).forEach(([type, pattern]) => {
+            const patternMatch = (new RegExp(pattern)).exec(url);
+            if (patternMatch) {
+                result[type] ??= [];
+                result[type].push({
+                    patternMatch,
+                    postExcerpt
+                });
+            }
+        });
+    }
+    return result;
+}
+
+const prefixWcsKey = (key) => ('wcs ' + key);
+
+async function getCommerceData(osi) {
+    const response = await fetch(wcsUrl(osi));
```

Comment:
Is that okay if the OSI doesn't return an offer anymore via WCS?  AOS can always deliver the parameters of an OSI.

Reply:
i think i prefer usage of WCS as blank answer there is also something we should be worried about